programs/account-compression/src/instructions/update_address_merkle_tree.rs

```diff
@@ -0,0 +1,81 @@
+use account_compression_state::{address_merkle_tree_from_bytes_mut, address_queue_from_bytes_mut};
+use anchor_lang::prelude::*;
+use ark_ff::BigInteger256;
+use light_indexed_merkle_tree::array::{IndexingElement, RawIndexingElement};
+use light_utils::be_bytes_to_bigint;
+
+use crate::{
+    errors::AccountCompressionErrorCode,
+    state::address::{AddressMerkleTreeAccount, AddressQueueAccount},
+};
+
+#[derive(Accounts)]
+pub struct UpdateMerkleTree<'info> {
+    #[account(mut)]
+    pub authority: Signer<'info>,
+    #[account(mut)]
+    pub queue: AccountLoader<'info, AddressQueueAccount>,
+    #[account(mut)]
+    pub merkle_tree: AccountLoader<'info, AddressMerkleTreeAccount>,
+}
+
+#[allow(clippy::too_many_arguments)]
+pub fn process_update_address_merkle_tree<'info>(
+    ctx: Context<'_, '_, '_, 'info, UpdateMerkleTree<'info>>,
+    // Index of the Merkle tree changelog.
+    changelog_index: u16,
+    // Index of the address to dequeue.
+    queue_index: u16,
+    // Index of the next address.
+    address_next_index: u16,
```

Comment:
Next address index needs to be a bigger number since it has to hold at least the number of leaves.
I think u64 is best here.

For changelog_index and queue_index u16 is ok since it's constrained to the size of the changelog and queue.

Reply:
Changed to usize here, it seems to work.

Also, I made the index fields in `IndexingArray` and `IndexingElement` generic. You might get a heart attack when looking at how many traits I had to use, but welp, it works and it's probably still better than copying the whole struct definitions or using macros.